RecoHI/HiEvtPlaneAlgos/src/EvtPlaneFilter.cc

```diff
@@ -0,0 +1,75 @@
+/*
+ * =====================================================================================
+ *       Filename:  EvtPlaneFilter.cc
+ *    Description:  Event plane Q2 filter
+ *        Created:  05/11/15 14:37:29
+ *         Author:  Quan Wang
+ * =====================================================================================
+ */
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDFilter.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+#include "RecoHI/HiEvtPlaneAlgos/interface/HiEvtPlaneList.h"
+#include "DataFormats/HeavyIonEvent/interface/EvtPlane.h"
+
+#include <iostream>
+
+class EvtPlaneFilter : public edm::EDFilter {
+public:
+	explicit EvtPlaneFilter(const edm::ParameterSet&);
+	~EvtPlaneFilter();
+private:
+	virtual void beginJob();
+	virtual bool filter(edm::Event&, const edm::EventSetup&);
+	virtual void endJob() ;
+
+	double vnlow_;
+	double vnhigh_;
+	int epidx_;
+	int eplvl_;
+	edm::Handle<reco::EvtPlaneCollection> ep_;
```

Comment:
remove ep_ from data members: it changes every event and is needed only in one function